elytron/WFCORE-4362-configurable-certificate-authority-in-certificate-authority-account.adoc

```diff
@@ -0,0 +1,91 @@
+= Make the certificate authority used by a certificate-authority-account configurable
+:author:            Farah Juma, Diana Vilkolakova
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, Let's Encrypt (https://acme-v02.api.letsencrypt.org/directory) is the only certificate authority
+that can be used in a `certificate-authority-account` resource.
+This issue aims to make it possible for the user to configure attribute `certificate-authority` for this resource.
+Each certificate authority can be associated with a production url and a staging url, both of which have to be configurable.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4362[WFCORE-4362]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1150[EAP7-1150]
+* https://issues.jboss.org/browse/ELY-1797[ELY-1797]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* mailto:mkopecky@redhat.com[Marek Kopecky]
+
+=== Testing By
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Elytron
+
+== Requirements
+
+=== Hard Requirements
+
+It must be possible to specify staging url and production url for the attribute `certificate-authority` in the `certificate-authority-account` resource.
+This will be achieved by adding new resource `certificate-authority` to the elytron subsystem.
+
+Eg. the configuration will look like the following:
+
+```
+/subsystem=elytron/certificate-authority=Buypass:add(url="https://api.buypass.com/acme/directory",staging-url="https://api.staging-example.com/acme/directory")
+/subsystem=elytron/certificate-authority-account=exampleCertificateAuthorityAccount:add(certificate-authority=Buypass,alias=example,key-store=exampleKS)
+```
+
+== Implementation Plan
+
+Currently, `certificate-authority-account` contains optional attribute `certificate-authority` of type STRING.
+The default and only allowed value is "LetsEncrypt".
+We should be able to specify staging URL and production URL without breaking backwards compatibility.
+
+In the Elytron subsystem, LetsEncrypt is a value of CertificateAuthority enum. This enum contains name, url and staging url.
+We can add new `certificate-authority` resource to the elytron
+subsystem with non-optional attributes `name`, `url` and optional attribute `staging-url`.
```

Comment:
Just a minor comment: `name` is probably not an attribute of `certificate-authority` resource, right? I'm trying your implementation:
```
[standalone@localhost:9990 /] /subsystem=elytron/certificate-authority=Buypass:read-attribute(name=<TAB>
staging-url  url  
[standalone@localhost:9990 /] /subsystem=elytron/certificate-authority=Buypass:read-attribute(name=name)
{
    "outcome" => "failed",
    "failure-description" => "WFLYCTL0201: Unknown attribute 'name'",
    "rolled-back" => true
}

[standalone@localhost:9990 /] 
```

Reply:
@marekkopecky In your example, name has value "Buypass". When you define certificate-authority in the standalone.xml, it looks like this:
```xml
<certificate-authorities>
    <certificate-authority name="Buypass" url="http://www.test.com"/>
</certificate-authorities>
```
I think CLI in this case is searching through attributes of resource Buypass specifically, which does not contain additional `name` attribute.

Reply:
@Skyllarr Ok, thank you for the explanation.